message/multipart.cpp

```diff
@@ -98,11 +102,17 @@ void Multipart::appendMultipart( EString &r, bool avoidUtf8 ) const
 
         Bodypart * bp = it;
         ++it;
-
-        r.append( bp->header()->asText( avoidUtf8 ) );
-        r.append( crlf );
-        appendAnyPart( r, bp, ct, avoidUtf8 );
-
+        
+        if ( isSigned ) {
+            ++it;       // skip next part, we append it raw
+            isSigned = false;
+            // just append our raw text, header is contained
+            appendAnyPart( r, bp, ct, avoidUtf8 );
```

Comment:
I think this still doesn't do the right thing. To test, sign this:

    Content-Type: text/plain
    Content-Transfer-Encoding: base64

    Zm9vYmFyCg==

The IMAP command `x uid search body foobar` should find that, but I expect that when the message is signed, it's not included in the search results.